torch/csrc/jit/script/module.cpp

```diff
@@ -358,7 +358,7 @@ void Module::train(bool on) {
   if (auto slot = find_attribute("training")) {
     slot->setValue(on);
   } else {
-    register_attribute("training", BoolType::get(), on);
+    AT_ERROR("'training' attribute not found");
```

Comment:
we shouldn't use `AT_ERROR` in new code, it should either be `TORCH_INTERNAL_ASSERT` or `TORCH_CHECK` (if user facing)